### ISSUE TITLE
validator - add timestamp checks

### ISSUE THREAD

**Issue by cwhanse** [SolarArbiter/solarforecastarbiter-core#18]
Check for gaps, irregular time steps, daylight savings transition.

Check timezone against clearsky and sunrise/sunset.

**Comment by cwhanse** [comment]
> Check for gaps, irregular time steps, daylight savings transition.

With the decision to require equally-spaced data, do we want a function in validator to check for timestamp spacing? Or will that be handled elsewhere?

**Comment by wholmgren** [comment]
That is mostly supposed to be handled in the API: https://github.com/SolarArbiter/solarforecastarbiter-api/issues/45

The API will only check for equally-spaced data in each data upload package. But there could be gaps created between data uploads. For example:

```
# upload 1
timestamp,value,quality_flag
2019-03-01T12:00:00Z,5,0
2019-03-01T12:05:00Z,,1
2019-03-01T12:10:00Z,5,0
2019-03-01T12:15:00Z,NaN,1
2019-03-01T12:20:00Z,NULL,1

# upload 2
timestamp,value,quality_flag
2019-04-01T12:00:00Z,5,0
2019-04-01T12:05:00Z,,1
2019-04-01T12:10:00Z,5,0
2019-04-01T12:15:00Z,NaN,1
2019-04-01T12:20:00Z,NULL,1
```

**Comment by wholmgren** [comment]
closed by #49
